lnwallet/btcwallet/btcwallet.go

```diff
@@ -1319,15 +1313,15 @@ func (b *BtcWallet) PublishTransaction(tx *wire.MsgTx, label string) error {
 	// `PublishTransaction` again because we need to mark the label in the
 	// wallet. We can remove this exception once we have the above TODO
 	// fixed.
-	case lnutils.ErrorAs[*base.ErrInMempool](err),
-		lnutils.ErrorAs[*base.ErrAlreadyConfirmed](err):
+	case errors.Is(err, rpcclient.ErrTxAlreadyInMempool),
```

Comment:
Nit: ` we'd always get ErrInMempool` is outdated above in the comment section.

Moreover I wonder why we mix here errors from the `testmempoolaccept` and `sendrawtransaction` cmd, don't we just have the former here as an errorr ?

Reply:
Updated the docs!

When an error is returned from `testmempoolaccept`, it's returned directly without any mapping. Here we just map the reject reason, and weirdly bitcoind gives different error messages for the same error case when calling `testmempoolaccept` vs `sendrawtransaction`.

Reply:
Ok in the `btcd` package you reference the error `ErrTxAlreadyConfirmed` to origin from `sendrawtransaction` so do you really need to check for it here when we only test for `testmempoolaccept ` ?

Reply:
oh that's what you meant, yeah we don't need to check it, updated. And for references:
https://github.com/bitcoin/bitcoin/blob/3d216baf91ca754e46e89788205513a956ec6d0a/test/functional/rpc_rawtransaction.py#L438-L439

Reply:
actually we still need this error mapping as `btcd` would return it here.

Reply:
ohh you are right, thank you